src/snowflake/connector/connection.py

```diff
@@ -1115,34 +1138,35 @@ def __config(self, **kwargs):
             ]:
                 self._authenticator = auth_tmp
 
-        if not self.user and self._authenticator != OAUTH_AUTHENTICATOR:
-            # OAuth Authentication does not require a username
-            Error.errorhandler_wrapper(
-                self,
-                None,
-                ProgrammingError,
-                {"msg": "User is empty", "errno": ER_NO_USER},
-            )
+        if not self._master_token and not self._session_token:
```

Comment:
why do we need check to master and session token here?

Reply:
In my understanding: Master and session token being provided should override the other authentication methods. 

So we check that if it's not provided, then we do the normal authentication workflow like it was before any code change.

Reply:
in this case, should the condition be `if not (self._master_token and self._session_token)` in which case we only skip the normal authentication workflow when both tokens are detected?

otherwise if only one token is provided, we shall use the normal authentication workflow

Reply:
Thanks for catching that, that makes sense to me.